datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java

```diff
@@ -304,6 +310,11 @@ private void reportMetrics(DatastreamRecordMetadata metadata, long eventsSourceT
       _dynamicMetricsManager.createOrUpdateSlidingWindowHistogram(MODULE, _datastreamTask.getConnectorType(),
           EVENTS_LATENCY_MS_STRING, LATENCY_SLIDING_WINDOW_LENGTH_MS, sourceToDestinationLatencyMs);
 
+      if (_enablePerTopicEventLatencyMetrics && !_enablePerTopicMetrics) {
```

Comment:
I didn't get this check. Can you please elaborate?

Reply:
If someone enables per topic metrics for all metrics via the `_enablePerTopicMetrics`, then I don't want to double count, since this the `topicOrDatastreamName` will point to the topic name already. Would you like me to add a comment?

Reply:
Please add a comment. It's not obvious. Thanks for the explanation

Reply:
done